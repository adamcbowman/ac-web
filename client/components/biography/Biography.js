import React from 'react'
import PropTypes from 'prop-types'
import CSSModules from 'react-css-modules'
import Avatar from '~/components/avatar'
import { Mailto, Phone } from '~/components/anchors'
import styles from './Biography.css'

// FIXME: This component is asking for too much! Find a way to be more generic. Look at Blocquote.

Biography.propTypes = {
    fullName: PropTypes.string.isRequired,
    email: PropTypes.string,
    title: PropTypes.string,
    avatar: PropTypes.string,
    workPhoneNumber: PropTypes.string,
    children: PropTypes.node,
}

function Biography({
    email,
    workPhoneNumber,
    children,
    fullName,
    title,
    avatar,
}) {
    return (
        <section styleName="Container">
            <div styleName="Media">
                <Avatar url={avatar} name={fullName} />
            </div>
            <div styleName="Content">
                <header styleName="Header">
                    <span styleName="Name">
                        {fullName}
                    </span>
                    {title &&
                        <span styleName="Title">
                            {title}
                        </span>}
                </header>
                <address styleName="Address">
                    {email &&
                        <Mailto
                            email={email}
                            title={`Send ${fullName} an email`}
                        />}
                    {workPhoneNumber &&
                        <Phone
                            phone={workPhoneNumber}
                            ext={null}
                            title={`Give ${fullName} a call`}
                        />}
                </address>
                {children}
            </div>
        </section>
    )
}

export default CSSModules(Biography, styles)
