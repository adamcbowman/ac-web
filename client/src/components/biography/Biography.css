@import "../../styles/colors.css";

.Container {
    color: var(--gray);
    margin: 5px;
    padding: 5px;
    line-height: 1.5em;
}

.Header {
    font-weight: bold;
}

.Name {
    color: var(--brand-primary);
}

.Title {
    margin-left: 5px;
    color: var(--gray-dark);
}

.Media {
    height: 100%;
    float: left;
    margin-right: 10px;
}
