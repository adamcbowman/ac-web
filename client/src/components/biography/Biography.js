import React, { PropTypes } from 'react'
import CSSModules from 'react-css-modules'
import styles from './Biography.css'
import { Image } from '../media'

Biography.propTypes = {
    name: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    avatar: PropTypes.string.isRequired,
    children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
}

function Biography({ name, title, avatar, children }) {
    return (
        <section styleName='Container'>
            <div styleName='Media'>
                <Image src={avatar}></Image>
            </div>
            <header styleName='Header'>
                <span styleName='Name'>{name}</span>
                <span styleName='Title'>{title}</span>
            </header>
            {children}
        </section>
    )
}

export default CSSModules(Biography, styles)
