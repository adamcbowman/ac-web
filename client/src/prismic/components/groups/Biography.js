import React, { PropTypes } from 'react'
import {compose, setPropTypes, mapProps} from 'recompose'
import {Biography} from 'components/biography'

export default compose(
    setPropTypes({
        document: PropTypes.object.isRequired,
    }),
    mapProps(({document}) => ({
        
    })),
)(Biography)
