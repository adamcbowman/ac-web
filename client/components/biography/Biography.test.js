import React from 'react'
import Renderer from 'react-test-renderer'
import Biography from './'

const Karl = {
    firstName: 'Karl',
    lastName: 'Guillotte',
    email: 'kguillotte@avalanche.ca',
    workPhoneNumber: '250-837-2141',
    ext: '112',
    title: 'Web developer',
    avatar: 'https://avatars1.githubusercontent.com/u/744011?v=3&s=60',
}

test('biography component', () => {
    const biography = Renderer.create(
        <Biography {...Karl}>Karl's bio</Biography>
    )

    expect(biography).toMatchSnapshot()
})
