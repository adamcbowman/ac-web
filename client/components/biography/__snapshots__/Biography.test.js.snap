// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`biography component 1`] = `
<section
  className="Container"
>
  <div
    className="Media"
  >
    <div
      className="Initials"
      data-initials="KG"
      style={
        Object {
          "fontSize": "1em",
          "height": 60,
          "width": 60,
        }
      }
    >
      <img
        alt="KG"
        onError={[Function]}
        onLoad={[Function]}
        src="https://avatars1.githubusercontent.com/u/744011?v=3&s=60"
        title="Karl Guillotte"
      />
    </div>
  </div>
  <div
    className="Content"
  >
    <header
      className="Header"
    >
      <span
        className="Name"
      >
        Karl Guillotte
      </span>
      <span
        className="Title"
      >
        Web developer
      </span>
    </header>
    <address
      className="Address"
    >
      <a
        href="mailto:kguillotte@avalanche.ca"
        title="Send Karl Guillotte an email"
      >
        kguillotte@avalanche.ca
      </a>
      <a
        href="tel:250-837-2141"
        title="Give Karl Guillotte a call"
      >
        250-837-2141
      </a>
    </address>
    Karl's bio
  </div>
</section>
`;
